great_expectations/expectations/expectation.py

```diff
@@ -104,6 +104,17 @@
 P = ParamSpec("P")
 T = TypeVar("T", List[RenderedStringTemplateContent], RenderedAtomicContent)
 
+try:
+    from typing_extensions import dataclass_transform
```

Comment:
This is interesting. I didn't know about this decorator. I'm going to read more about this shortly. Thanks!

Reply:
It's new as of python 3.11
https://peps.python.org/pep-0681/